### ISSUE TITLE
Dropout

### ISSUE THREAD

**Issue by severous** [Diego999/pyGAT#55]
if we set `self.dropout=True`, the probability of an element to be zeroed is one. then all elements of  output tenor is zero.
I get confused about it
```python
class GAT(nn.Module):
    def __init__(self, nfeat, nhid, nclass, dropout, alpha, nheads):
        """Dense version of GAT."""
        super(GAT, self).__init__()
        self.dropout = dropout

        self.attentions = [GraphAttentionLayer(nfeat, nhid, dropout=dropout, alpha=alpha, concat=True) for _ in range(nheads)]
        for i, attention in enumerate(self.attentions):
            self.add_module('attention_{}'.format(i), attention)

        self.out_att = GraphAttentionLayer(nhid * nheads, nclass, dropout=dropout, alpha=alpha, concat=False)

    def forward(self, x, adj):
        x = F.dropout(x, self.dropout, training=self.training)
        x = torch.cat([att(x, adj) for att in self.attentions], dim=1)
        x = F.dropout(x, self.dropout, training=self.training) # if dropout is true,x is zero
        x = F.elu(self.out_att(x, adj))
        return F.log_softmax(x, dim=1)
```

**Comment by Gemaxis** [comment]
--dropout', type=float, default=0.6
